modules/product/src/main/java/com/opengamma/strata/product/fx/Barrier.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.fx;
+
+import java.time.LocalDate;
+
+/**
+ * Definition of barrier event of option instruments. 
+ * <p>
+ * The barrier type, knock type, barrier level and other relevant information are specified in this class. 
```

Comment:
The barrier level can be date dependent. For forward starting barrier, the barrier level can be set to very high or low level in the initial period.